aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -3063,10 +3068,7 @@ Tensor& linalg_svdvals_out(const Tensor& A, Tensor & S) {
 }
 
 Tensor linalg_svdvals(const Tensor& A) {
-  const bool A_requires_grad = (at::GradMode::is_enabled() && A.requires_grad())
-                               || A._fw_grad(/*level */ 0).defined()
-                               || isTensorSubclassLike(A);
```

Comment:
@Lezcano functorch still seems to pass without `isTensorSubclassLike` here

Reply:
Hmmm. Some tests from functorch (something like "batched over jvp/vjp" or so) did not pass a week ago when we tested this. This was because the wrapper tensor did not have `requires_grad=True` nor a forward grad. What changed that makes it work now?

Reply:
That is weird... here are some more data points:

Without `isTensorSubclassLike`:
- Without the forward grad defined check I run into `RuntimeError: U_.dim() >= 2 && Vh_.dim() >= 2INTERNAL ASSERT FAILED at "../torch/csrc/autograd/FunctionsManual.cpp":2391, please report a bug to PyTorch.` - for `test_jvp`, `test_vmapjvp` and `test_vmapjvpall` (linalg_svdvals_cpu_float32).
- With the forward grad defined check, all 3 tests pass.

With `isTensorSubclassLike`:
- all 3 pass with or without forward grad defined check

To repro:
- run something like `python test/test_ops.py -k test_jvp_linalg_svdvals`

Reply:
This PR only changes the testing side, but with the updated testing I found that `eigvals` and `maxpool_1d` needed similar changes to get the tests to pass. `eigvalsh` seemed like a similar case, so I also updated that.

Reply:
I mean, I'm happy if the tests pass. We just added that condition to make some tests pass, but if those are green I'm happy for this condition to go.

But again, it's odd, as when we debugged this, we saw that the external tensor did not have `requires_grad=True`, so it was reasonable for the test to fail. Let's just hope that it does not fail later on, and if it does, that `INTERNAL_ASSERT` will fail and we'll know what to revert :P